CharlieHewitt/IndustrialTeamProject
# Backend timer setup + Lobby refactor + working request game loop prototype

There are a bunch of TODO:'s for tomorrow. (some can be found scattered throughout the code)
- I will update the API documentation now.

Tested as follows:

Create Lobby -> Join Lobby -> host/settings (3 questions (1 per category)) -> host/start -> start -> nextQuestion -> polling/questionOver -> polling/leaderboardOver -> nextQuestion -> questionOver -> leaderboardOver .... -> Quiz finished.

If anybody wants to help out testing the backend API using Postman I have a library of API requests that I used, with valid requests that just need the correct lobbyId, playerId and/or questionNumber added in. I can share the file containing these on slack tomorrow if you ask me.

Still to be added/tested: 
- /answer + /skip + /fiftyfifty + /leaderboard (They have clear timing windows defined for them (also phases) will mention this tomorrow.

- /nextQuestion should notify the user if they ask for 'the wrong question'

- thoroughly test routes for wrong LobbyId/ playerId ...

- implement only host can use host/settings or host/start.

- add updating scores logic to start of leaderboard phase.

- integrate with frontend by calling the above requests/ polling at the correct times according to phases. I will do this with @justasam tomorrow.

- removing a lobby once the game has ended

- redeploy backend once it works as intended

- deploy frontend

- remove/comment out obsolete routes

- if there's time left ... tests

- update Readme.md to link to everybody's GitHub profiles from their names

- report
